File: C++/source/Regression/RegressionTestEntry.cpp
/***************************************************************************
 *   Copyright (C) 2005 Vidyasagara Guntaka                                *
 *   All rights reserved.                                                  *
 *   Author : Vidyasagara Reddy Guntaka                                    *
 ***************************************************************************/

#include "Regression/RegressionTestEntry.h"

namespace WaveNs
{
RegressionTestEntry::RegressionTestEntry ()
    : m_testServiceId (0),
      m_isTestEnabled (false)
{
}

RegressionTestEntry::RegressionTestEntry (WaveServiceId id, bool enabled)
    : m_testServiceId (id),
      m_isTestEnabled (enabled)
{
}

RegressionTestEntry::~RegressionTestEntry ()
{
}

WaveServiceId RegressionTestEntry::getTestServiceId () const
{
    return (m_testServiceId);
}

bool RegressionTestEntry::getIsTestEnabled () const
{
    return (m_isTestEnabled);
}

void RegressionTestEntry::setIsTestEnabled (bool enabled)
{
    m_isTestEnabled = enabled;
}

}

File: Java/source/com/CxWave/Wave/Regression/RegressionTestEntry.java
/*************************************************************************************************************************
 * Copyright (C) 2015-2016 Vidyasagara Guntaka & CxWave, Inc * All rights reserved. * Author : Vidyasagara Reddy Guntaka *
 *************************************************************************************************************************/

package com.CxWave.Wave.Regression;

import com.CxWave.Wave.Framework.Type.WaveServiceId;

public class RegressionTestEntry
{
    private final WaveServiceId m_testServiceId;
    private final boolean       m_isTestEnabled;

    public RegressionTestEntry (final WaveServiceId testServiceId, final boolean isTestEnabled)
    {
        m_testServiceId = testServiceId;
        m_isTestEnabled = isTestEnabled;
    }

    public WaveServiceId getTestServiceId ()
    {
        return (m_testServiceId);
    }

    public boolean getIsTestEnabled ()
    {
        return (m_isTestEnabled);
    }
}
